blocks/extensions/saveload.py

```diff
@@ -85,83 +84,78 @@ def do(self, callback_name, *args):
         construction stage.
 
         """
-        from_main_loop, from_user = self.parse_args(callback_name, args)
+        _, from_user = self.parse_args(callback_name, args)
         try:
             path = self.path
             if from_user:
                 path, = from_user
-            already_saved_to = self.main_loop.log.current_row.get(SAVED_TO, ())
-            self.main_loop.log.current_row[SAVED_TO] = (
-                already_saved_to + (path,))
-            secure_pickle_dump(self.main_loop, path)
+            secure_dump(self.main_loop, path)
```

Comment:
Because for dumping the main loop, I want to use the custom `dump` function we wrote (which writes everythong to a ZIP file). However, for the attributes (normally the log and the iteration state) we can just use `cPickle.dump`, which is faster, and easier for a user to unpack (because anything serialized with our `dump` function, can only be loaded with the custom `load` function).